airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -66,26 +64,38 @@ def request_params(
         if next_page_token:
             return next_page_token
         else:
-            return {"page[size]": self.page_size}
+            return {"page[size]": self.page_size} if self.page_size else {}
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         """:return an iterable containing each record in the response"""
+
         response_json = response.json()
         for record in response_json.get("data", []):  # API returns records in a container array "data"
             record = self.map_record(record)
             yield record
 
-    def map_record(self, record: Mapping):
+    def map_record(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
         """Subclasses can override this to apply custom mappings to a record"""
+
+        record[self.cursor_field] = record["attributes"][self.cursor_field]
         return record
 
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
```

Comment:
I have added it to the main class because the implementation is the same for both `IncrementalKlaviyoStream` and `SemiIncrementalKlaviyoStream` - just to avoid duplicated code. Do you think this is wrong approach here?